components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultRequestCoordinator.java

```diff
@@ -390,6 +394,18 @@ && isStepHasMultiOption(context)
                     }
                 }
 
+                if (isPromptRequest(request)) {
+                    AuthGraphNode currentNode =
+                            (AuthGraphNode) context.getProperty(FrameworkConstants.JSAttributes.PROP_CURRENT_NODE);
+                    // If the request is a prompt request returning from browser's back button, we need to handle it
+                    // by changing context to previous step.
+                    if (!(currentNode instanceof ShowPromptNode)) {
+                        currentNode = moveToShowPromptNode(currentNode);
+                        context.setCurrentStep(context.getCurrentStep() - 1);
```

Comment:
Can there be a scenario where this values will be < 0

Reply:
Fixed in https://github.com/wso2/carbon-identity-framework/pull/6647/commits/ffb0217e864a5bf0248aea542ae14d3e57475659